state-chain/pallets/cf-elections/src/benchmarking.rs

```diff
@@ -253,43 +253,28 @@ mod benchmarks {
 	fn provide_shared_data() {
 		let validator_id = ready_validator_for_vote::<T, I>(1)[0].clone();
 
-		let (election_identifier, ..) =
-			Pallet::<T, I>::electoral_data(&validator_id.clone().into())
-				.unwrap()
-				.current_elections
-				.into_iter()
-				.next()
-				.unwrap();
+		let shared_data_value: <
+			<T::ElectoralSystemRunner as ElectoralSystemTypes>::VoteStorage as VoteStorage
+		>::SharedData = BenchmarkValue::benchmark_value();
 
-		assert_ok!(Pallet::<T, I>::vote(
-			RawOrigin::Signed(validator_id.clone()).into(),
-			Box::new(
-				BoundedBTreeMap::try_from(
-					[(
-						election_identifier,
-						AuthorityVoteOf::<T::ElectoralSystemRunner>::Vote(
-							BenchmarkValue::benchmark_value()
-						),
-					)]
-					.into_iter()
-					.collect::<BTreeMap<_, _>>(),
-				)
-				.unwrap()
-			),
-		));
+		let shared_data_hash = SharedDataHash::of(&shared_data_value);
 
-		#[extrinsic_call]
-		provide_shared_data(
-			RawOrigin::Signed(validator_id),
-			Box::new(BenchmarkValue::benchmark_value()),
+		SharedDataReferenceCount::<T, I>::insert(
+			shared_data_hash,
+			UniqueMonotonicIdentifier::from(0),
+			ReferenceDetails::<BlockNumberFor<T>> {
+				count: 1u32,
+				created: BlockNumberFor::<T>::from(1u32),
+				expires: BlockNumberFor::<T>::from(10u32),
+			},
 		);
 
-		assert_eq!(
-			SharedData::<T, I>::get(SharedDataHash::of::<
-				VoteOf<<T as Config<I>>::ElectoralSystemRunner>,
-			>(&BenchmarkValue::benchmark_value())),
-			Some(BenchmarkValue::benchmark_value())
-		);
+		ElectionConsensusHistoryUpToDate::<T, I>::insert(UniqueMonotonicIdentifier::from(0), 0);
+
+		#[extrinsic_call]
+		provide_shared_data(RawOrigin::Signed(validator_id), Box::new(shared_data_value.clone()));
+
+		assert!(SharedData::<T, I>::get(shared_data_hash).is_some());
```

Comment:
I think we could also check that the shared data we get is the one we provided, as we did previously, no?